File: types/skill.ts
export type Category = {
  id: string;
  slug: string;
  name: string;
  icon: string;
};

export type Skill = {
  id: string;
  slug: string;
  name: string;
  description: string;
  description_zh: string | null;
  category: string;
  tags: string[];
  use_cases: string[];
  author: string;
  github_url: string;
  skill_md_content: string | null;
  skill_md_summary_zh: string | null;
  published_at: string | null;
  github_stars: number | null;
  rank: number;
  score: number;
  featured: boolean;
  created_at: string;
  updated_at: string;
};

export type SkillListItem = Pick<
  Skill,
  | "id"
  | "slug"
  | "name"
  | "description"
  | "description_zh"
  | "category"
  | "tags"
  | "use_cases"
  | "author"
  | "github_stars"
  | "rank"
  | "score"
  | "featured"
  | "created_at"
  | "published_at"
>;

File: Skiller/Models/Skill.swift
import Foundation

struct Skill: Codable, Identifiable, Hashable {
    let id: String
    let slug: String
    let name: String
    let description: String
    let descriptionZh: String?
    let category: String
    let tags: [String]
    let useCases: [String]?
    let useCasesEn: [String]?
    let author: String
    let githubUrl: String
    let githubStars: Int?
    let rank: Int
    let score: Int
    let featured: Bool
    let createdAt: String
    let publishedAt: String?
    let skillMdContent: String?
    let installCount: Int?

    enum CodingKeys: String, CodingKey {
        case id, slug, name, description, category, tags, author, rank, score, featured
        case descriptionZh    = "description_zh"
        case useCases         = "use_cases"
        case useCasesEn       = "use_cases_en"
        case githubUrl        = "github_url"
        case githubStars      = "github_stars"
        case createdAt        = "created_at"
        case publishedAt      = "published_at"
        case skillMdContent   = "skill_md_content"
        case installCount     = "install_count"
    }
}

struct SkillCategory: Codable, Identifiable, Hashable {
    let id: String
    let slug: String
    let name: String
    let icon: String
}

struct RepoGroupSummary: Codable, Identifiable, Hashable {
    var id: String { repo }
    let repo: String
    let author: String
    let stars: Int?
    let skillCount: Int
    let repSkillId: String

    enum CodingKeys: String, CodingKey {
        case repo, author, stars
        case skillCount = "skill_count"
        case repSkillId = "rep_skill_id"
    }
}

struct HomeStats {
    let total: Int
    let newToday: Int
    let lastSyncAt: Date?
}

extension Skill {
    /// Picks Chinese description for zh users, English for everyone else.
    /// Falls back to whichever exists when the preferred one is missing/empty.
    var localizedDescription: String {
        let prefersChinese = Locale.current.language.languageCode?.identifier == "zh"
        if prefersChinese, let zh = descriptionZh, !zh.isEmpty {
            return zh
        }
        if !description.isEmpty {
            return description
        }
        return descriptionZh ?? ""
    }

    /// Locale-appropriate use-case chips: English for non-zh systems, Chinese
    /// otherwise. Falls back to the other language if the preferred one hasn't
    /// been generated yet (older rows in the DB).
    var localizedUseCases: [String] {
        let prefersChinese = Locale.current.language.languageCode?.identifier == "zh"
        let preferred = prefersChinese ? useCases : useCasesEn
        if let p = preferred, !p.isEmpty { return p }
        let fallback = prefersChinese ? useCasesEn : useCases
        return fallback ?? []
    }
}
